learn/guides/distinctive-language-features/advanced-general-purpose-language-features.md

```diff
@@ -1,51 +1,58 @@
-description: Let’s now look at the other features of the Ballerina language. These are a mixed bag of additional options to the language, making everything fit together to build a Ballerina application.
+description: Let's now look at the other features of the Ballerina language. These are a mixed bag of additional options to the language, making everything fit together to build a Ballerina application.
 keywords: ballerina, programming language, ballerina packages,language-guide
 permalink: /learn/distinctive-language-features/advanced-general-purpose-language-features/
 active: advanced-general-purpose-language-features
-intro: Let’s now look at the other features of the Ballerina language. These are a mixed bag of additional options to the language, making everything fit together to build a Ballerina application.
+intro: Let's now look at the other features of the Ballerina language. These are a mixed bag of additional options to the language, making everything fit together to build a Ballerina application.
 redirect_from:
 - /learn/distinctive-language-features/advanced-general-purpose-language-features
 ---
 
 ## Default Values for Function Parameters
 
-Ballerina allows the setting of default values for function parameters. You can use literal values or even expressions as default values to parameters. Additionally, the default value expressions can use the values of preceding parameters.
+Ballerina allows specifying default values for function parameters. You can use any expression such as a literal or a function call as the default value of a parameter. Additionally, the default value expressions can use the values of preceding parameters.
 
 ```ballerina
-function substring(
-    string str, 
-    int start = 0,
-    int end = str.length()
-) returns int | ( ) {
-  //….
+function substring(string str, int 'start = 0, int end = str.length()) 
+            returns int? {
+    // ...
 }
 ```
 
-The way it works is, the type descriptor contains closures for every defaultable parameter, which has the values for previous parameters as parameters for those closures. Using those closures, the compiler generates code to fill missing values at the call site for this function. This is not part of the function type and is not applicable when functions are passed around as a first class value.
+The way it works is, the type descriptor contains closures for every defaultable parameter, which has the values for previous parameters as parameters for those closures. Using those closures, the compiler generates code to fill missing values at the call site for this function. This is not part of the function type and is not applicable when functions are passed around as a first-class value.
 
 ## Providing Function Arguments by Name
 
 In addition to default values for parameters, it is also useful to call functions with argument names. Ballerina allows you to call functions where arguments can be supplied by name.
```

Comment:
We can emphasize parameters instead of calling.
`We can provide values to parameters in any order using named arguments`